test/e2e/e2e_test.go

```diff
@@ -145,15 +145,17 @@ func main(m *testing.M) int {
 	}
 	defer alertSLOs(testContext)
 
-	oidcBucketName := "hypershift-ci-oidc"
-	iamClient := e2eutil.GetIAMClient(globalOpts.configurableClusterOptions.AWSCredentialsFile, globalOpts.configurableClusterOptions.Region)
-	s3Client := e2eutil.GetS3Client(globalOpts.configurableClusterOptions.AWSCredentialsFile, globalOpts.configurableClusterOptions.Region)
-	if err := setupSharedOIDCProvider(oidcBucketName, iamClient, s3Client); err != nil {
-		log.Error(err, "failed to setup shared OIDC provider")
-		return -1
-	}
-	defer e2eutil.DestroyOIDCProvider(log, iamClient, globalOpts.IssuerURL)
-	defer e2eutil.CleanupOIDCBucketObjects(log, s3Client, oidcBucketName, globalOpts.IssuerURL)
+	if globalOpts.Platform == hyperv1.AWSPlatform {
+		oidcBucketName := "hypershift-ci-oidc"
+		iamClient := e2eutil.GetIAMClient(globalOpts.configurableClusterOptions.AWSCredentialsFile, globalOpts.configurableClusterOptions.Region)
+		s3Client := e2eutil.GetS3Client(globalOpts.configurableClusterOptions.AWSCredentialsFile, globalOpts.configurableClusterOptions.Region)
+		if err := setupSharedOIDCProvider(oidcBucketName, iamClient, s3Client); err != nil {
+			log.Error(err, "failed to setup shared OIDC provider")
+			return -1
+		}
```

Comment:
this function generates the `globalOpts.ServiceAccountSigningKey = keyBytes`. when the signing key is not set, then the e2e test creates the signing key secret without any key, which causes the hosted cluster to fail to be reconciled.